### PR TITLE
Cleanup execution environment across wrappers and entrypoints

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Add a single `use_snap_env` helper function to configure the execution environment so that binaries and libraries take precedence over the ones from the host environment.

### Changes

- Add `use_snap_env` function to configure PATH, LD_LIBRARY_PATH, PYTHONPATH, LANG, LC_ALL, PYTHONNOUSERSITE, OPENSSL_CONF
- Cleanup all wrappers and replace ad-hoc logic with the helper util
- Change shebang from `/bin/bash` to `/usr/bin/env bash` to fix issues with foreign distributions.

### CODE REVIEW

**Comment by ktsakalozos** [comment]
@neoaggelos lets not forget to handle the LD_LIBRARY_PATH and PATH in the `sudo -E` calls.

File: microk8s-resources/wrappers/microk8s-join.wrapper [hunk 1]
```diff
@@ -21,7 +16,7 @@ exit_if_no_permissions
 
 if is_strict
 then
-  LD_LIBRARY_PATH=$IN_SNAP_LD_LIBRARY_PATH ${SNAP}/usr/bin/python3 ${SNAP}/scripts/wrappers/join.py "${@}"
+  ${SNAP}/usr/bin/python3 ${SNAP}/scripts/wrappers/join.py "${@}"
 else
-  sudo -E LD_LIBRARY_PATH=$IN_SNAP_LD_LIBRARY_PATH ${SNAP}/usr/bin/python3 ${SNAP}/scripts/wrappers/join.py "${@}"
+  sudo -E ${SNAP}/usr/bin/python3 ${SNAP}/scripts/wrappers/join.py "${@}"
```
Comment (ktsakalozos) [inline microk8s-resources/wrappers/microk8s-join.wrapper:21]: Don't we drop PATH and LD_LIBRARY_PATH here?

File: microk8s-resources/wrappers/microk8s-ctr.wrapper [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
+  "${SNAP}/bin/ctr" "${args[@]}" "$@"
 else
-  declare -a args="($(cat $SNAP_DATA/args/ctr))"
-  if is_strict
-  then
-    LD_LIBRARY_PATH="$IN_SNAP_LD_LIBRARY_PATH" CONTAINERD_SNAPSHOTTER="$SNAPSHOTTER" "${SNAP}/bin/ctr" "${args[@]}" "$@"
-  else
-    sudo -E LD_LIBRARY_PATH="$IN_SNAP_LD_LIBRARY_PATH" CONTAINERD_SNAPSHOTTER="$SNAPSHOTTER" "${SNAP}/bin/ctr" "${args[@]}" "$@"
-  fi
+  sudo -E "${SNAP}/bin/ctr" "${args[@]}" "$@"
```
Comment (ktsakalozos) [inline microk8s-resources/wrappers/microk8s-ctr.wrapper:24]: Don't we drop PATH and LD_LIBRARY_PATH here?

File: microk8s-resources/wrappers/microk8s-join.wrapper (hunk 1 above)
Comment (neoaggelos) [reply microk8s-resources/wrappers/microk8s-join.wrapper]: thanks, missed it. should be good now

File: microk8s-resources/wrappers/microk8s-ctr.wrapper (hunk 2 above)
Comment (neoaggelos) [reply microk8s-resources/wrappers/microk8s-ctr.wrapper]: dito

**Pull request merged**